utils.js

```diff
@@ -0,0 +1,25 @@
+const chars =
+	"ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz0123456789+/=";
+function atob(input: string = "") {
+	let str = input.replace(/=+$/, "");
+	let output = "";
+
+	if (str.length % 4 == 1) {
+		throw new Error(
+			"'atob' failed: The string to be decoded is not correctly encoded.",
+		);
+	}
+	for (
+		let bc = 0, bs = 0, buffer, i = 0;
+		(buffer = str.charAt(i++));
+		~buffer && ((bs = bc % 4 ? bs * 64 + buffer : buffer), bc++ % 4)
+			? (output += String.fromCharCode(255 & (bs >> ((-2 * bc) & 6))))
```

Comment:
This blows up the memory and consumes about 100 MiB.

Reply:
You are right. We can replace it with `Buffer.from(input, 'base64').toString('binary')`.

Reply:
We need this change for mobile for sure

Reply:
@Sekhmet Are you planning to make this change? not sure if I should wait or continue with brining the Monero PRs to master then make this change

Reply:
Sorry, I didn't know you are waiting for those changes before merging PRs. I can do it now.

Reply:
Please do, will audit your mobile PR tonight